pkg/keys/ecdsa.go

```diff
@@ -73,14 +65,87 @@ func (p *p256Verifier) UnmarshalPublicKey(key *data.PublicKey) error {
 		return err
 	}
 
-	curve := elliptic.P256()
+	ecdsaKey, ok := p.PublicKey.PublicKey.(*ecdsa.PublicKey)
+	if !ok {
+		return fmt.Errorf("invalid public key")
+	}
+	p.ecdsaKey = ecdsaKey
+	p.key = key
+	return nil
+}
+
+type ecdsaSigner struct {
+	*ecdsa.PrivateKey
+}
+
+type ecdsaPrivateKeyValue struct {
+	Private string         `json:"private"`
+	Public  *PKIXPublicKey `json:"public"`
+}
 
-	// Parse as uncompressed marshalled point.
-	x, _ := elliptic.Unmarshal(curve, p.PublicKey)
-	if x == nil {
-		return errors.New("tuf: invalid ecdsa public key point")
+func (s *ecdsaSigner) PublicData() *data.PublicKey {
+	keyValBytes, _ := json.Marshal(EcdsaVerifier{PublicKey: &PKIXPublicKey{PublicKey: s.Public()}})
```

Comment:
No checking for errors?

Reply:
Same as above, but in fact the entire key interface should hanlde errors in `Public()` and `PublicData()` where they currently return no errors.